immunopepper/ip.py

```diff
@@ -101,10 +101,22 @@ def parse_arguments(argv):
     so.add_argument("--output-count", help="request to write the intermediate number of kmer at each each step to the given path (risk of slow down)" , required=False, default='')
     so.add_argument("--expression-fields-c", nargs='+', help="name of segment and junction expression field in cancer file, default ['segment_expr', 'junction_expr']",required=False, default=None)
     so.add_argument("--tag-normals", help="name for the normal cohort output files, use when various normal cohorts", required=False, default='')
-    so.add_argument("--tag-prefix", help="prefix to use for the output files, use when several conditions", required=False, default='')
+    so.add_argument("--tag-prefix", help="prefix to use for the output files, use when several conditions", required=False, default=None)
     so.add_argument("--whitelist-normal", help="file containg whitelist for normal samples", required=False, default=None)
     so.add_argument("--whitelist-cancer", help="file containg whitelist for cancer samples", required=False, default=None)
-
+    so.add_argument("--annotated-flags", nargs='+', required=False, default=[],
+                    help="list with the annotation modes requested."
+                    "Mode C0: No filter based on junction or read frame annotation status in cancer samples"
+                    "Mode C1: In cancer samples filter for (junctionAnnotated == 1) & (readFrameAnnotated==1)"
+                    "Mode C2: In cancer samples filter for (junctionAnnotated == 1) & (readFrameAnnotated==0)"
+                    "Mode C3: In cancer samples filter for (junctionAnnotated == 0) & (readFrameAnnotated==1)"
+                    "Mode C4: In cancer samples filter for (junctionAnnotated == 0) & (readFrameAnnotated==0)"
+                    "Mode N0: No filter based on junction or read frame annotation status in normal samples"
+                    "Mode N1: In normal samples filter for (junctionAnnotated == 1) & (readFrameAnnotated==1)"
+                    "Mode N2: In normal samples filter for (junctionAnnotated == 1) & (readFrameAnnotated==0)"
+                    "Mode N3: In normal samples filter for (junctionAnnotated == 0) & (readFrameAnnotated==1)"
+                    "Mode N4: In normal samples filter for (junctionAnnotated == 0) & (readFrameAnnotated==0)"
+                    "e.g. [C1, C2, N2, N4] or [C1, C2, N0]. Default: ignores all flags")
```

Comment:
I assume that the user could want to get the **union** of different conditions on _junction annotated_ and _read frame annotated_ in a **single filtering run**. 
Therefore, in the current implementation this is a **OR operation**: For example [C3, N1, N2] would be 
On cancer side we want reading frame annotated on neojunctions: 

`Keep C3: ((junctionAnnotated == 0) & (readFrameAnnotated==1))`

On normal side, we want junction annotated regardless of the reading frame

`Keep N2 or N1: if ((junctionAnnotated == 1) & (readFrameAnnotated==0)) | ((junctionAnnotated == 1) & (readFrameAnnotated==1))
`

And [C3, N0] would be: 
On cancer side: 

`Keep C3: ((junctionAnnotated == 0) & (readFrameAnnotated==1))`

On normal side: 

`
Don't do any filtering
`

---

If the user does not want to do any filtering on cancer or normal, the user can either (1) Ignore the --annotated-flags parameter because the default is []. This is what I tried to mean by "Default: ignores all flags" (2) the user can also enter [C0, N0] 

---
It was a little bit of a headache to think about a user interface more efficient than having parameters for let's say 
```
--junctionAnnotated_in_normals
--readFrameAnnotated_in_normals
--junctionAnnotated_in_cancers
--readFrameAnnotated_in_cancers
```
 and running the filtering mode several times if the user wishes to get the union of the sets. 

---
With the explanation above I will wait for your general impression on the user interface before making corrections